fdbclient/MonitorLeader.actor.cpp

```diff
@@ -900,7 +907,23 @@ ACTOR Future<MonitorLeaderInfo> monitorProxiesOneGeneration(
 		state ClusterConnectionString storedConnectionString;
 		if (connRecord) {
 			bool upToDate = wait(connRecord->upToDate(storedConnectionString));
-			if (!upToDate) {
+			if (upToDate) {
+				incorrectTime = Optional<double>();
+			} else if (allConnectionsFailed) {
+				// Failed to connect to all coordinators from the current connection string,
+				// so it is not possible to get any new updates from the cluster. It can be that
+				// all the coordinators have changed, but the client missed that, because it had
+				// an incompatible protocol version. Since the cluster file is different,
+				// it may have been updated by other clients.
+				TraceEvent("UpdatingConnectionStringFromFile")
+				    .detail("ClusterFile", connRecord->toString())
+				    .detail("StoredConnectionString", storedConnectionString.toString())
+				    .detail("CurrentConnectionString", connRecord->getConnectionString().toString());
+				connRecord->setAndPersistConnectionString(storedConnectionString);
```

Comment:
This returns a future, so we should wait on it.

Reply:
I notice now that we aren't waiting on this function anywhere else. We get away with that because the only implementations we currently use of this record are synchronous, but there is one implementation that is asynchronous and I guess would break if we didn't wait on it.

Reply:
I have now changed all uses of setAndPersistConnectionString to wait on the future.